crates/mod-dns-resolver/src/lib.rs

```diff
@@ -145,8 +147,10 @@ pub fn register(lua: &Lua) -> anyhow::Result<()> {
                 });
             }
 
-            let resolver = TokioAsyncResolver::tokio(r_config, config.options);
-            dns_resolver::reconfigure_resolver(HickoryResolver::from(resolver));
+            let mut builder =
+                TokioResolver::builder_with_config(r_config, TokioConnectionProvider::default());
+            *builder.options_mut() = config.options;
```

Comment:
This API is arguably a little idiosyncratic, I think we'll improve on that (see https://github.com/hickory-dns/hickory-dns/pull/2877).